polaris-core/src/main/java/org/apache/polaris/core/persistence/pagination/DonePageToken.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.persistence.pagination;
+
+import java.util.List;
+
+public class DonePageToken extends PageToken {
+
+  public DonePageToken() {}
+
+  @Override
+  public String toString() {
+    return null;
+  }
+
+  @Override
+  protected PageToken updated(List<?> newData) {
+    return new DonePageToken();
```

Comment:
It is an error for a "done" token to have a next page, right? Should we throw an exception here?

Reply:
I think so, but it'll depend how the caller uses it. I'll make the change for now and we can figure out the particulars of some of these methods when we actually implement pagination.